Automattic/wp-calypso
# Add of Migration Trials and initial refactoring of Ecommerce Trial components

This initial commit will cover the addition of Migration Trial and refactoring of WooExpress Trials to become generic "trial" components.

The PR does not cover 100% of components to keep the PR size down. The next changes are on https://github.com/Automattic/wp-calypso/pull/80141 (draft).

This is the list of the components that are used:

1. `CurrentPlan`
    - `ECommerceTrialCurrentPlan`
       - `ECommerceTrialBanner`
       - List of features included
       - Ready to start selling?
       - List of features not included
2. `ECommerceTrialExpired` (not changed right now)
       - `TrialUpgradeConfirmation` (not changed right now)
3. `Plans`
     - `ECommerceTrialPlansPage` (not changed right now)

Additional: p1690534487000689-slack-C01A60HCGUA

Fixes https://github.com/Automattic/wp-calypso/issues/80076
Fixes https://github.com/Automattic/wp-calypso/issues/79955

## Proposed Changes

* Move the trial checkout URL functions to a generic `client/lib/trials/get-*-trial-checkout-url.ts`
* Add support in the `CurrentPlan` component for `PLAN_MIGRATION_TRIAL_MONTHLY` plans

New components and refactoring:
1. `CurrentPlan`
    - `TrialCurrentPlan` <- moved to `client/my-sites/plans/current-plan/trials/trial-current-plan.tsx` and renamed
       - `TrialBanner`
       - Replace the list with new components `EcommerceTrialIncluded`, `MigrationTrialIncluded`
       - Ready to start selling?
       - Replace the list with new components `EcommerceTrialNotIncluded`, `MigrationTrialNotIncluded`
4. `ECommerceTrialExpired` (not changed right now)
      - `TrialUpgradeConfirmation` (not changed right now)
5. `Plans`
     - `ECommerceTrialPlansPage` (not changed right now)

## Notes
1. The migration current plan page show buttons in the boxes. In the ECommerce one, they are hidden by default.
2. We have no additional section similar to the "Ready to start selling?" on the ECommerce page. So the Migration page right now shows nothing.
3. "Jetpack backups and restores" CTA is TBD now. I've used "View your backup activity" for now (not localized).

These are the URLs that are right now associated with the CTAs:

1. "Beautiful Themes: **Browse themes**":  `/themes/__URL__`
2. "Advanced Design Tools: **Design your blog**": `wp-admin/site-editor.php`
5. "Newsletter: **Setup a newsletter**":  `/setup/newsletter/intro?siteSlug=__SLUG__`
6. "Jetpack backups and restores: **View your backup activity**": `/backup/__SLUG__`
7. "Spam Protection: **Keep your site safe**": `https://jetpack.com/blog/what-is-spam/`
8. "SEO tools: **Increase visibility**": `https://wordpress.com/support/seo-tools/`
9. "Google Analytics: **Connect Google Analytics**": `https://wordpress.com/support/google-analytics/`

## Testing Instructions



* Create a JN blog, install "Move to WP" and from `localhost` and start the free plan
* Create a blog from WooCommerce site and choose a free plan
* Visit `http://calypso.localhost:3000/plans/my-plan/__JN_URL__`
* Visit `http://calypso.localhost:3000/plans/my-plan/__WOOEXPRESS_URL__`
* Visit `https://wordpress.com/plans/my-plan/__WOOEXPRESS_URL__`

* Ensure that the Migration Trial page reflects the [Figma file](https://href.li/?https://www.figma.com/file/nRzixgCz1TsnOLPiKr4awj/Free-trials-in-the-Migrate-to-WP.com-flow?type=design&node-id=1-64&mode=design&t=SHxI569VmRWJ1Aaf-0)
* Ensure that the WooCommerce page is the same found in production

![screenshot](https://github.com/Automattic/wp-calypso/assets/167611/8254a31c-df61-4b1d-87ef-96277472ebb3)

## ECommerce final part
<img width="1077" alt="image" src="https://github.com/Automattic/wp-calypso/assets/167611/b41c839c-6f49-4665-a900-99a0857cb14e">

## Pre-merge Checklist
